### PR TITLE
Add more JGit / netrc related debugging

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/core/src/main/kotlin/OrtAuthenticator.kt [hunk 1]
```diff
@@ -129,19 +130,22 @@ internal fun getNetrcAuthentication(contents: String, machine: String): Password
 
     val iterator = lines.joinToString(" ").split(Regex("\\s+")).iterator()
 
-    var machineFound = false
+    var machineFound: String? = null
```
Comment (fviernau) [inline utils/core/src/main/kotlin/OrtAuthenticator.kt:133]: How about naming this variable 'machine'?

File: utils/core/src/main/kotlin/OrtAuthenticator.kt [hunk 2]
```diff
@@ -129,19 +130,22 @@ internal fun getNetrcAuthentication(contents: String, machine: String): Password
...
+            "login" -> login = machineFound?.let { iterator.nextOrNull() }
+            "password" -> password = machineFound?.let { iterator.nextOrNull() }
+            "default" -> machineFound = token
         }
 
-        if (login != null && password != null) return PasswordAuthentication(login, password.toCharArray())
+        if (login != null && password != null) {
+            OrtAuthenticator.log.debug { "Found a '$machineFound' entry for '$machine'." }
+            return PasswordAuthentication(login, password.toCharArray())
+        }
```
Comment (fviernau) [inline utils/core/src/main/kotlin/OrtAuthenticator.kt:148]: In order to allow a bit more detailed debugging, how instead logging in lines 140 / 141 like the following?

"Found login $login for machine $machine."
"Found a password for machine $machine."

**Review by fviernau — APPROVED** [review]

File: utils/core/src/main/kotlin/OrtAuthenticator.kt (hunk 1 above)
Comment (sschuberth) [reply utils/core/src/main/kotlin/OrtAuthenticator.kt:133]: That would conflict with the function parameter of the same name.

File: utils/core/src/main/kotlin/OrtAuthenticator.kt (hunk 2 above)
Comment (sschuberth) [reply utils/core/src/main/kotlin/OrtAuthenticator.kt:148]: I didn't want to expose that much detail yet. Might be, though, that I'm forced to in a follow-up, if the current additions do not suffice for my current debugging.

File: utils/core/src/main/kotlin/OrtAuthenticator.kt (hunk 1 above)
Comment (fviernau) [reply utils/core/src/main/kotlin/OrtAuthenticator.kt:133]: I missed that, thanks!

**Pull request merged**